services/grpc.go

```diff
@@ -0,0 +1,115 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net"
+	"strings"
+
+	"github.com/kacperf531/sockchat/api"
+	pb "github.com/kacperf531/sockchat/protobuf"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type GrpcAPI struct {
+	pb.UnimplementedSockchatServer
+	core        *SockchatCoreService
+	authService *SockchatAuthService
+}
+
+func NewSockchatGRPCServer(core *SockchatCoreService, authService *SockchatAuthService) *grpc.Server {
+	grpcApi := &GrpcAPI{core: core, authService: authService}
+	server := grpc.NewServer(grpc.UnaryInterceptor(grpcApi.AuthInterceptor))
+	pb.RegisterSockchatServer(server, grpcApi)
+	return server
+}
+
+var protectedRequests = map[string]bool{
+	"GetProfile":        true,
```

Comment:
```suggestion
var methodAuthorization = map[string]bool{
	"RegisterProfile":   false,
	"GetProfile":        true,
```
optional: you could list all the gRPC methods to explicitly include/exclude authorization for all of them